operator/apis/platform/v1alpha1/central_types.go

```diff
@@ -153,20 +153,20 @@ func (c *CentralComponentSpec) GetAdminPasswordGenerationDisabled() bool {
 	return pointer.BoolPtrDerefOr(c.AdminPasswordGenerationDisabled, false)
 }
 
-// CentralDBEnabled returns true if central db should be created
-// TODO(ROX-13557): modify the logic to enable Postgres DB by default.
-func (c *CentralComponentSpec) CentralDBEnabled() bool {
-	if c == nil || c.DB == nil || c.DB.IsEnabled == nil {
+// IsExternalDB returns true if central DB is not managed by the Operator
+func (c *CentralComponentSpec) IsExternalDB() bool {
+	if c == nil || c.DB == nil {
 		return false
 	}
-
-	return *c.DB.IsEnabled == CentralDBEnabledTrue
+	return c.DB.IsExternal()
 }
 
 // CentralDBSpec defines settings for the "central db" component.
+// TODO(ROX-14395): drop `IsEnabled` field.
 type CentralDBSpec struct {
-	// Specify whether central-db is enabled, Default configures central to use rocksdb.
-	// If IsEnabled the operator will provision the Central DB or use an external DB if a connection string is provided.
+	// Deprecated field. It is no longer necessary to specify it.
+	// This field will be removed in a future release.
+	// Central configures to use PostgreSQL by default.
 	//+kubebuilder:validation:Default=Default
 	//+kubebuilder:default=Default
 	//+operator-sdk:csv:customresourcedefinitions:type=spec,order=1
```

Comment:
We probably want to hide it in UI now that it has no effect.

```suggestion
	//+operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:hidden"}
```
This will also require renumbering the order of fields below.